cachito/workers/scm.py

```diff
@@ -192,7 +197,10 @@ def update_and_archive(self, previous_archive, gitsubmodule=False):
                 # The reference must be specified to handle commits which are not part
                 # of a branch.
                 repo.remote().fetch(refspec=self.ref)
-            except:  # noqa E722
+
+            except Exception as ex:
+                template = "An exception of type {0} occurred. Arguments:\n{1!r}"
+                log.exception(template.format(type(ex).__name__, ex.args))
```

Comment:
IMO, we don't need to repeat the `log.exception`. The existing `log.exception` is able to log traceback which has the exception type info.

Regarding the `ex.args`, I also don't think we need it, since what the value would be does totally depend on the concrete exception implementation. No all are valuable for developers to debug. Again, the following `log.exception` already logs the `self.url`. In addition, if we also care about the ref for the fetch operation, we should log the `self.ref` explicitly.

 Similar thought applies to other changes as well in this pr.

Reply:
@cqi  +1 for adding template.format(type(ex).__name__  info into original `log.exception` (merge two log.exceptions here into one). I don't mind adding `ex.args` into log message. It might or might not be useful. I rather get more info. +1 fro adding `self.ref` explicitly into log.

Reply:
@fepas Are we going to add also `self.ref` to log message?

Reply:
Nice, thanks!